bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -138,16 +150,21 @@ impl ClientBuilder {
         // Determine server either from URL, server name or user ID.
         if let Some(homeserver_url) = builder.homeserver_url {
             inner_builder = inner_builder.homeserver_url(homeserver_url);
-        } else if let Some(server_name) = builder.server_name {
+        } else if let Some((server_name, protocol)) = builder.server_name {
             let server_name = ServerName::parse(server_name)?;
-            inner_builder = inner_builder.server_name(&server_name);
+            let protocol = match protocol.as_str() {
+                "http" => ServerNameProtocol::Http,
+                "https" => ServerNameProtocol::Https,
+                _ => anyhow::bail!("invalid protocol {protocol}"),
+            };
```

Comment:
Instead of letting the user specify an arbitrary string and erroring if it's not `http` or `https`, could we make the public interface be `client_builder.insecure_unencrypted_http()` / `client_builder.insecure_no_tls()` or somesuch?

Reply:
This is for internal use only, to be fair (only defined within the context of that crate), so I'll make use of an enum instead.